scatter-gather/scatter-gather.bal

```diff
@@ -0,0 +1,50 @@
+import ballerina/lang.regexp;
+import ballerina/http;
+
+type CarRental record {|
+    string carType;
+    string price;
+    string resultLink;
+|};
+
+type AvisResponse record {
+    record {
+        *CarRental;
+    }[] result;
+};
+
+service /api/v1 on new http:Listener(8080) {
+    
+    resource function get vehicles(string dropOffDateTime, string dropOffLocation, string pickupDateTime,
+    string pickupLocation) returns map<CarRental[]>|error {
+    
+        worker hotwire returns CarRental[]|error {
+            http:Client hotwireEP = check new ("http://api.hotwire.com.balmock.io");
+            http:Response hotwireResponse = check hotwireEP -> /v1/search/car.get(pickup=pickupLocation, 
+            dest=dropOffLocation, startDate=pickupDateTime, endDate=dropOffDateTime);
+            return transformHotwireResponse(check hotwireResponse.getXmlPayload());
+        }
+
+        worker avis returns CarRental[]|error {
+            http:Client avisEP = check new ("http://api.avis.com.balmock.io");
+            AvisResponse avisResponse = check avisEP -> /cars/catalog/v1/vehicles/rates.get(brand="Avis", 
+            country_code="US", dropOff_date=dropOffDateTime, dropoff_location=dropOffLocation,
+            pickup_date=pickupDateTime, pickup_location=pickupLocation);
+            return avisResponse.result.'map(r => {carType: r.carType, price: r.price, resultLink: r.resultLink});
+        }
+
+        return map from CarRental[]|error carRental in wait {hotwire, avis} 
+                where carRental is CarRental[] && carRental.length() > 0
+                let string vendor = (<regexp:Span>(<regexp:Groups[]> re `/(\w+)\.`
+                    .findAllGroups(carRental[0].resultLink))[0][1]).substring()
+                select [vendor, carRental];
```

Comment:
Can we do something simpler eg:
```
var r = check wait {hotwire, avis} ;
return [...check r.hotwire, ...check r.avis];
``

Reply:
We cannot use this approach. If we want to get rid of the error part we need to iterate through the responses.